### DISCUSSION TITLE
How to use quorum Explorer

### DISCUSSION THREAD

**Discussion by 0721z** [ConsenSys/quorum#1445] [Q&A] [answered]
I have created a private network, and I want to use quorum explorer to view the situation of the network and carry out operations such as publishing contracts and transactions. But there was an error when I ran quorum explorer.

hangzv@zhangzv-MS-7C88:~/桌面/quorum-explorer$ npm run dev

> quorum-explorer-next@0.2.2 dev
> next dev -p 25000

ready - started server on 0.0.0.0:25000, url: http://localhost:25000
info  - Loaded env from /home/zhangzv/桌面/quorum-explorer/.env.local
warn  - You have enabled experimental feature(s).
warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use them at your own risk.

Bus error (core dumped)

Here is my config.json
{
  "algorithm": "qbft",
  "nodes": [
   {
      "name": "Node-0", 
      "client": "goquorum", 
      "rpcUrl": "http://0.0.0.0:22000", 
      "privateTxUrl": "" 
   },
    {
      "name": "Node-1",
      "client": "goquorum",
      "rpcUrl": "http://0.0.0.0:22001",
      "privateTxUrl": ""
   },
   {
      "name": "Node-2",
      "client": "goquorum",
      "rpcUrl": "http://0.0.0.0:22002",
      "privateTxUrl": ""
   },
   {
      "name": "Node-3",
      "client": "goquorum",
      "rpcUrl": "http://0.0.0.0:22003",
      "privateTxUrl": ""
   },
   {
      "name": "Node-4",
      "client": "goquorum",
      "rpcUrl": "http://0.0.0.0:22004",
      "privateTxUrl": ""
   }
   ]
}

The following is the log file when I started
Jul  6 11:47:01 zhangzv-MS-7C88 sudo:  zhangzv : TTY=pts/1 ; PWD=/home/zhangzv/桌面/quorum-explorer ; USER=root ; COMMAND=/usr/local/bin/npm run dev
Jul  6 11:47:01 zhangzv-MS-7C88 sudo: pam_unix(sudo:session): session opened for user root by (uid=0)
Jul  6 11:47:02 zhangzv-MS-7C88 sudo: pam_unix(sudo:session): session closed for user root
Can you provide a detailed tutorial on configuring quorum explore? I thank you in advance for your help.

**Comment by baptiste-b-pegasys** [comment, marked as answer]
Hello, I think you are referring to this project : https://github.com/ConsenSys/quorum-explorer

There is an example in the quick dev start, from the docker compose you can see:
```docker
  explorer:
    image: consensys/quorum-explorer:e3ac210
    volumes:
      - ./quorum-explorer/config.json:/app/config.json
      - ./quorum-explorer/env:/app/.env.production
    depends_on:
      - rpcnode
    ports:
      - 25000:25000/tcp
    networks:
      quorum-dev-quickstart:
        ipv4_address: 172.16.239.31
```

from the example you are giving, it is correct for validators, but for members you need to add private information (if you want support for private contracts)

```json
{
      "name": "member1",
      "client": "goquorum",
      "rpcUrl": "http://member1quorum:8545",
      "privateTxUrl": "http://member1tessera:9080",
      "privateKey": "0x8f2a55949038a9610f50fb23b5883af3b4ecb3c3bb792cbcefbd1542c692be63",
      "accountAddress": "0xfe3b557e8fb62b89f4916b721be55ceb828dbd73"
},
```
